core/lib/types/src/fee.rs

```diff
@@ -74,8 +74,14 @@ impl Fee {
         assert!(block_base_fee_per_gas <= self.max_fee_per_gas);
         assert!(self.max_priority_fee_per_gas <= self.max_fee_per_gas);
 
-        // For now, we charge only for base fee.
-        block_base_fee_per_gas
+        // The following calculation is based on the EIP-1559 formula:
+        // https://eips.ethereum.org/EIPS/eip-1559
+        let priority_fee_per_gas = std::cmp::min(
+            self.max_priority_fee_per_gas,
+            self.max_fee_per_gas - block_base_fee_per_gas,
+        );
```

Comment:
Are you sure it's a correct fix? There are other places where we also calculate `effective_gas_price`, e.g. [here](https://github.com/lambdaclass/zksync-era/blob/fix-effective-gas-price-calculation/core/lib/multivm/src/versions/vm_boojum_integration/tracers/refunds.rs#L120). It feels a bit suspicious that you only change a single place for that.

cc @StanislavBreadless @perekopskiy -- do you guys have any feedback on this?